modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/display/context/DLItemSelectorViewDisplayContext.java

```diff
@@ -364,8 +368,39 @@ private Long _getFileEntryTypeId() {
 		InfoItemItemSelectorCriterion infoItemItemSelectorCriterion =
 			(InfoItemItemSelectorCriterion)_itemSelectorCriterion;
 
-		return GetterUtil.getLong(
-			infoItemItemSelectorCriterion.getItemSubtype());
+		long fileEntryTypeId = GetterUtil.getLong(
+			infoItemItemSelectorCriterion.getItemSubtype(), -1);
+
+		DLFileEntryType dlFileEntryType = null;
+
+		if (fileEntryTypeId >= 0) {
+			dlFileEntryType = _dlFileEntryTypeLocalService.fetchFileEntryType(
+				fileEntryTypeId);
+		}
+
+		if (dlFileEntryType == null) {
+			fileEntryTypeId = _getFileEntryTypeId(
+				infoItemItemSelectorCriterion.getItemSubtype());
+		}
+
+		return fileEntryTypeId;
```

Comment:
I would simplify this piece of code. I don't see the need to fetch the FileEntyType by id.
```java
		long fileEntryTypeId = GetterUtil.getLong(
			infoItemItemSelectorCriterion.getItemSubtype(), -1);

		if (fileEntryTypeId < 0) {
			fileEntryTypeId = _getFileEntryTypeId(
				infoItemItemSelectorCriterion.getItemSubtype());
		}

		return fileEntryTypeId;
```

Reply:
It is because the infoItemItemSelectorCriterion.getItemSubtype() is represent as a String, but it can be a number, so in this case we should discard this option where the number represents the FileEntryTypeId or the FileEntryTypeKey

Reply:
You mean a FileEntryTypeKey can also be represented as a number? Is that right @adolfopa ?

Reply:
Yes FileEntryTypeKey can also represent a number, by the way I've already simplified the logic